code/stable/glassbr/src/cpp/InputParameters.cpp

```diff
@@ -236,6 +238,20 @@ void InputParameters::input_constraints() {
         std::cout << "." << std::endl;
         throw("InputError");
     }
+    if (!(set<double>{2.5, 2.7, 3.0, 4.0, 5.0, 6.0, 8.0, 10.0, 12.0, 16.0, 19.0, 22.0}.find(this->t) != set<double>{2.5, 2.7, 3.0, 4.0, 5.0, 6.0, 8.0, 10.0, 12.0, 16.0, 19.0, 22.0}.end())) {
```

Comment:
This set shouldn't be inlined 3 times. The "error" here is in the specification, it should name this set. This shouldn't be done automatically by the framework which is likely doing exactly what it is being asked to do.

In fact, we should probably make it an error to output a concrete value of a compound type in this way.

Reply:
I know we discussed in #3869 what would be involved to create a variable above the `if`-statement.  The types and control-flow involved make it harder than it probably should be, but it's definitely doable.

Reply:
I agree that this very hard to do post-facto, i.e. in GOOL itself. It should be easier to do in `drasil-code` and potentially should be done even further upstream than that.

Basically: sets should never be anonymous values. But we shouldn't fix that "just too late" but ask that whoever uses sets, does it 'right'.

Reply:
Sets are now defined in a variable before being used